scripts/resolve-version.js

```diff
@@ -0,0 +1,68 @@
+#!/usr/bin/env node
+const path = require('path');
+const fs = require('fs');
+
+const ALLOWED_RELEASE_TYPES = [ 'alpha', 'rc', 'stable' ];
+const ROOTDIR = path.resolve(__dirname, '..');
+
+//
+// parse release.json
+//
+
+const releaseFile = 'release.json';
+const releaseConfig = require(path.join(ROOTDIR, releaseFile));
+const majorVersion = releaseConfig.majorVersion;
+const releaseType = releaseConfig.releaseType;
+if (!majorVersion) { throw new Error(`"majorVersion"" must be defined in ${releaseFile}`); }
+if (!releaseType) { throw new Error(`"releaseType" must be defined in ${releaseFile}`); }
+if (majorVersion !== '1' && majorVersion !== '2') { throw new Error(`majorVersion=${majorVersion} is an unsupported major version (should be a number)`); }
```

Comment:
"a number" seems to be vastly more permissive than "1 or 2".

Reply:
Changed.